fendermint/app/src/cmd/genesis.rs

```diff
@@ -280,6 +289,19 @@ fn set_ipc_gateway(genesis_file: &PathBuf, args: &GenesisIpcGatewayArgs) -> anyh
     })
 }
 
+async fn seal_genesis(genesis_file: &PathBuf, args: &SealGenesisArgs) -> anyhow::Result<()> {
+    let genesis = read_genesis(genesis_file)?;
+
+    let genesis_creator = GenesisCreator::new(
+        args.builtin_actors_path.clone(),
+        args.custom_actors_path.clone(),
+        args.artifacts_path.clone(),
+        args.output_path.clone(),
+    );
+
+    genesis_creator.create(genesis).await
```

Comment:
These APIs are weird. We have a Genesis and the GenesisCreator. You would imagine the GenesisCreator creates a Genesis, but in reality the Genesis already exists and the GenesisCreator wants it as an input.

I think we need to adjust this a little.
- We should have a `GenesisBuilder`.
- The `genesis_file` is poorly named. These are genesis parameters, so let's name it `genesis_params_path`.
- You can then have a `GenesisBuilder::from_params` to construct a new `GenesisBuilder` from the parameters file.
- We then have a method for each artifact we need to take: `with_builtin_actors`, `with_custom_actors`, `with_system_contracts`.
- We then have a `write_into<W: io::Write>(w: &mut W)` method that writes the genesis file to a writer.
- The file is managed by `seal_genesis`.

In practice:

```rust
let mut output = // create the file

let gen = GenesisBuilder::from_params(genesis_params_path)
    .with_builtin_actors(...)
    .with_custom_actors(...)
    .with_system_contracts(...)
    .write_into(&output)
```

Reply:
This works for me. Just a side note that probably 
```
.with_builtin_actors(...)
.with_custom_actors(...)
.with_system_contracts(...)
```
should be added to constructor as they are required parameters, otherwise it feels it's not really required.